lib/components/JoinWallet.js

```diff
@@ -24,39 +24,49 @@ class JoinWallet extends PureComponent {
     return {
       selectedWallet: PropTypes.string,
       joinWallet: PropTypes.func,
+      history: PropTypes.object, // react router history
     };
   }
 
   static get defaultProps() {
-    return {};
+    return {
+      selectedWallet: '',
+    };
   }
 
   update(key, value) {
     this.setState({ [key]: value });
   }
 
+  navigate(path) {
+    const { history } = this.props;
+    history.push(path);
+    history.goForward();
+  }
+
   async join() {
     const { joinWallet, selectedWallet } = this.props;
     const { cosignerId, selectedHardware, joinKey, accountNumber } = this.state;
 
-    const options = {
-      walletId: selectedWallet,
+    await joinWallet(selectedWallet, {
       cosignerId,
       joinKey,
       account: accountNumber,
       hardwareType: selectedHardware,
-    };
-
-    await joinWallet(options);
+    });
 
-    // TODO: redirect to multisig-complete page
-    // to display cosigner token
+    const path = `/bwallet/multisig-complete/${selectedWallet}`;
```

Comment:
should the base, `bwallet`, come from either the `metadata`, react router itself, or the namespace variable (first two seem the best options)? This feels brittle.

Reply:
Good call, it definitely is brittle, I was just trying to get it working. I'm not even sure if i like the path name being called `multisig-complete`, but I have 1 page that it goes to after creating a multisig wallet or joining a multisig wallet to display secrets. Do you have any suggestions as to what the path could be called? I'll define it in `constants` and use it everywhere that needs it

Reply:
would `multisig-join` make sense? or maybe `multisig/join` which feels a little more restful. It could even be `multisig/complete` which allows you to do something like `[base]/:type/complete/:id` and even `complete` could be a variable param, like `:state` or `:status`. Kind of just spitballing here though :D

Reply:
I like `/multisig/:action` so that the same page could be used for both join and init